projects/kit/components/input-slider/input-slider.template.html

```diff
@@ -65,22 +65,41 @@
                 [mode]="hintController.mode"
                 [showDelay]="hintController.showDelay"
                 [hideDelay]="hintController.hideDelay"
-                (mousedown.prevent)="onMouseDown()"
+                (mousedown.prevent)="focusTextInput()"
             ></tui-tooltip>
         </span>
     </div>
 </tui-wrapper>
-<tui-slider
-    class="t-slider"
-    [min]="min"
-    [max]="max"
-    [steps]="computedSteps"
-    [segments]="segments"
-    [keySteps]="keySteps"
-    [pluralize]="segmentsPluralize"
-    [focusable]="false"
-    [disabled]="readOnly || disabled"
-    [ngModel]="value"
-    (ngModelChange)="onSliderValue($event)"
-    (mousedown)="onMouseDown()"
-></tui-slider>
+
+<!--TODO: remove wrapper + ticks labels (leave only slider) in v3.0-->
+<div class="t-slider-wrapper">
+    <input
+        tuiSlider
+        type="range"
+        [tuiFocusable]="false"
+        [max]="computedSteps"
+        [segments]="segments"
+        [keySteps]="computedKeySteps"
+        [attr.disabled]="readOnly || disabled || null"
+        [ngModel]="value"
+        (ngModelChange)="onSliderValue($event)"
+        (click)="focusTextInput()"
+    />
+
+    <!--TODO delete segments labels in v3.0-->
+    <div *ngIf="segmented && !isNew" class="t-segments-labels">
+        <span
+            *tuiRepeatTimes="let segmentIndex of segments + 1"
+            automation-id="tui-slider__segment"
+        >
+            <span *ngIf="fromToTexts$ | async as fromToText">
+                {{ getSegmentPrefix(segmentIndex, fromToText) }} {{
+                getSegmentLabel(segmentIndex) | tuiFormatNumber }}
```

Comment:
`getSegmentLabel` is pure function? will there be performance problem here?

Reply:
why not 

```
{{ fromToText | segmentPrefix: segmentIndex }} {{ segmentIndex | segmentLabel | tuiFormatNumber }}


```

Reply:
It is just copy-paste from old code))

But I agree, pipes is suitable for this.
I'll make it

Reply:
and this help to optimize imports... amazing!
thanks!

Reply:
I have made pipe `tuiSliderTickLabel` and marked `getSegmentPrefix` with `@tuiPure` decorator